### PR TITLE
WIP: Bugfix/navigation

### PR DESCRIPTION
**Pull request by feruzm**
**TODO**

- [x] Navigation issues when no user is logged in. Check Login to see button, click doesn't respond.

- [x] Click on profile icon bottom nav, it will open login screen, click cancel and app crashes (Android)

- [x] Open post and click back (hardware or software), you will see blank white screen (Android)

- [x] Investigate this commit, it might be cause: https://github.com/esteemapp/esteem-mobile/commit/1043af2b0452c1037f01fd15cb9ef33e57837f5f#diff-1fdf421c05c1140f6d71444ea2b27638

- [x] Other small bug fixes

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -46,9 +47,8 @@
     "moment": "^2.22.2",
     "react": "16.9.0",
     "react-intl": "^3.3.1",
-    "react-native": "0.61.2",
+    "react-native": "^0.61.5",
```
Comment (mistikk) [inline package.json]: please revert to rn version, we should follow this guide for upgrade https://github.com/react-native-community/rn-diff-purge/compare/release/0.61.2..release/0.61.5

File: package.json [hunk 2]
```diff
@@ -60,14 +60,14 @@
...
-    "react-native-keyboard-aware-scroll-view": "^0.8.0",
+    "react-native-keyboard-aware-scroll-view": "^0.9.1",
     "react-native-linear-gradient": "^2.4.2",
     "react-native-modal-dropdown": "^0.7.0",
     "react-native-modal-popover": "0.0.12",
     "react-native-navigation-bar-color": "^0.1.0",
     "react-native-reanimated": "^1.3.0",
     "react-native-render-html": "^4.1.2",
-    "react-native-screens": "^2.0.0-alpha.16",
+    "react-native-screens": "^2.0.0-alpha.18",
```
Comment (mistikk) [inline package.json]: please revert to this version update. because i removed this library with #1394

File: src/components/bottomTabBar/view/bottomTabBarStyles.js [hunk 3]
```diff
@@ -6,7 +6,7 @@ const deviceWidth = Dimensions.get('window').width;
 
 export default EStyleSheet.create({
   wrapper: {
-    width: '$deviceWidth',
+    width: deviceWidth,
```
Comment (mistikk) [inline src/components/bottomTabBar/view/bottomTabBarStyles.js]: can you remove `const deviceWidth = Dimensions.get('window').width;` instead of this changes

File: src/components/bottomTabBar/view/bottomTabBarView.js [hunk 4]
```diff
@@ -40,6 +40,7 @@ const BottomTabBarView = ({
   const dispatch = useDispatch();
 
   useEffect(() => {
+    console.log('updateActiveBottomTab', routes[index].routeName);
```
Comment (mistikk) [inline src/components/bottomTabBar/view/bottomTabBarView.js]: please remove console codes

File: src/components/postCard/view/postCardView.js [hunk 5]
```diff
@@ -113,6 +113,7 @@ class PostCardView extends Component {
         </View>
         <View style={styles.postBodyWrapper}>
           <TouchableOpacity
+            // eslint-disable-next-line react-native/no-inline-styles
             style={[{ flexDirection: 'column' }]}
```
Comment (mistikk) [inline src/components/postCard/view/postCardView.js]: can you move to styles file

File: src/screens/application/container/applicationContainer.js [hunk 6]
```diff
@@ -604,7 +599,6 @@ class ApplicationContainer extends Component {
 
     if (isConnected !== null && isConnected !== nextProps.isConnected && nextProps.isConnected) {
       this._fetchApp();
-      this.globalInterval = setInterval(this._refreshGlobalProps, 180000);
```
Comment (mistikk) [inline src/screens/application/container/applicationContainer.js]: it was global props refreshing, did you test it without this

File: src/utils/wallet.js [hunk 7]
```diff
@@ -41,17 +41,15 @@ export const groomingTransactionData = (transaction, steemPerMVests, formatNumbe
...
-        ',',
-        '.',
-      );
-      steemPayout = formatNumber(parseToken(steemPayout), { minimumFractionDigits: 3 }).replace(
-        ',',
-        '.',
-      );
-      vestingPayout = formatNumber(vestsToSp(parseToken(vestingPayout), steemPerMVests), {
-        minimumFractionDigits: 3,
-      }).replace(',', '.');
```
Comment (mistikk) [inline src/utils/wallet.js]: i guess we was use `formatNumber ` for a issue, i couldnt remember what was the issue. Why did you remove to that

File: package.json (hunk 1 above)
Comment (feruzm) [reply package.json]: There is no guide, it is difference between version and I checked changelog, only bugs are fixed, no major changes, that's why I upgraded.

File: package.json (hunk 2 above)
Comment (feruzm) [reply package.json]: ok!

File: src/components/bottomTabBar/view/bottomTabBarStyles.js (hunk 3 above)
Comment (feruzm) [reply src/components/bottomTabBar/view/bottomTabBarStyles.js]: `deviceWidth` is being used in Line 25, that's why I kept it and changed about to local deviceWidth as well so that we don't have to travel back to get it from global. If you can suggest fix for Line 25, then we can completely remove Line 5

File: src/components/bottomTabBar/view/bottomTabBarView.js (hunk 4 above)
Comment (feruzm) [reply src/components/bottomTabBar/view/bottomTabBarView.js]: ok

File: src/components/postCard/view/postCardView.js (hunk 5 above)
Comment (feruzm) [reply src/components/postCard/view/postCardView.js]: ok!

File: src/screens/application/container/applicationContainer.js (hunk 6 above)
Comment (feruzm) [reply src/screens/application/container/applicationContainer.js]: We do `_fetchApp()` which refreshes variables on startup right? So we don't need interval, everytime user restarts app, fetchApp() updates that data and it should be sufficient.

File: src/utils/wallet.js (hunk 7 above)
Comment (feruzm) [reply src/utils/wallet.js]: On Android Nexus S API 24, numbers were not correctly formatted, long numbers in curation and author rewards. These was reported by users multiple times and I couldn't find root cause other than formatNumber not working properly on older Android devices, so it is simple Floating point change, so `.toFixed()` should do the job just fine.

**Comment by feruzm** [comment]
@mistikk Please review again, especially `sideMenuView.js`, I converted to react hooks, let me know if I did it correctly.

File: package.json (hunk 1 above)
Comment (mistikk) [reply package.json]: yes it is not guide but offical documents recommend to this way. i know it not has any major update but if we not apply all changes we can encounter to some issues on feature updates

File: src/components/bottomTabBar/view/bottomTabBarStyles.js (hunk 3 above)
Comment (mistikk) [reply src/components/bottomTabBar/view/bottomTabBarStyles.js]: hmm okay i didnt see that. i just reviewed 25. line, we cannot change to that line so you can use it is okay

File: src/screens/application/container/applicationContainer.js (hunk 6 above)
Comment (mistikk) [reply src/screens/application/container/applicationContainer.js]: yes i agree with you but we discuss to this before, if user not kill to application it will not refresh

File: src/utils/wallet.js (hunk 7 above)
Comment (mistikk) [reply src/utils/wallet.js]: hmm okay we can try this solution

File: src/components/sideMenu/view/sideMenuView.js [hunk 8]
```diff
@@ -20,53 +20,46 @@ import styles from './sideMenuStyles';
...
+  const [isAddAccountIconActive, setIsAddAccountIconActive] = useState(false);
+  const [storageT, setStorageT] = useState('R');
 
   // Component Life Cycles
-  componentDidMount() {
+  useEffect(() => {
+    let _isMounted = false;
     getStorageType().then(item => {
-      this.setState({ storageT: item });
+      if (!_isMounted) {
```
Comment (mistikk) [inline src/components/sideMenu/view/sideMenuView.js]: i guess this if is not necessary, it will always true. maybe you wanted to check item

File: src/components/sideMenu/view/sideMenuView.js [hunk 9]
```diff
@@ -20,53 +20,46 @@ import styles from './sideMenuStyles';
...
+    setIsAddAccountIconActive(!isAddAccountIconActive);
   };
 
-  _handleOnMenuItemPress = item => {
-    const { navigateToRoute, switchAccount } = this.props;
-
+  const _handleOnMenuItemPress = item => {
     if (item.id === 'logout') {
+      // eslint-disable-next-line react/no-this-in-sfc
       this.ActionSheet.show();
```
Comment (mistikk) [inline src/components/sideMenu/view/sideMenuView.js]: you cannot use this keyword outside of react.component. if you want to ref with react hooks, it has useRef() func. if you can search on all repo you can see usage

File: src/components/sideMenu/view/sideMenuView.js [hunk 10]
```diff
@@ -78,113 +71,109 @@ class SideMenuView extends Component {
...
-                  <IconButton
-                    name={isAddAccountIconActive ? 'ios-arrow-dropup' : 'ios-add-circle-outline'}
-                    color="white"
-                    size={20}
-                    onPress={this._handleOnPressAddAccountIcon}
-                    style={styles.addAccountIcon}
+    setMenuItems(isLoggedIn ? MENU.AUTH_MENU_ITEMS : MENU.NO_AUTH_MENU_ITEMS);
+
+    return () => {};
+  }, []);
```
Comment (mistikk) [inline src/components/sideMenu/view/sideMenuView.js]: you should put isLoggedIn to this array. Because it was componentWillReceiveProps, it is mean this function was call every props or states changed. but default useEffect function just calling on componenDidMount. If you want to call every props and state changes, you should put this key to last array. So it should be [isLoggedIn]

File: src/components/sideMenu/view/sideMenuView.js [hunk 11]
```diff
@@ -78,113 +71,109 @@ class SideMenuView extends Component {
...
+
+  //UNSAFE_componentWillReceiveProps(nextProps) {}
```
Comment (mistikk) [inline src/components/sideMenu/view/sideMenuView.js]: you can remove to this

File: src/components/sideMenu/view/sideMenuView.js [hunk 12]
```diff
@@ -20,53 +20,46 @@ import styles from './sideMenuStyles';
...
-    this.state = {
-      menuItems: props.isLoggedIn ? MENU.AUTH_MENU_ITEMS : MENU.NO_AUTH_MENU_ITEMS,
-      isAddAccountIconActive: false,
-      storageT: 'R',
-    };
-  }
+const SideMenuView = ({
+  currentAccount,
+  isLoggedIn,
+  intl,
```
Comment (mistikk) [inline src/components/sideMenu/view/sideMenuView.js]: you can not get intl with this. you can use useIntl() hooks

**Comment by mistikk** [comment]
@feruzm i reviewed, you can check last comments

File: package.json (hunk 1 above)
Comment (feruzm) [reply package.json]: How to apply all changes? I thought yarn fetches all updates with you add new version. Let me know steps and I will do it.

File: src/components/sideMenu/view/sideMenuView.js (hunk 8 above)
Comment (feruzm) [reply src/components/sideMenu/view/sideMenuView.js]: It is necessary in some cases, function might be async and component might be unmounted already. But, I will fix it, it should have unsubscribe `return`.

File: src/components/sideMenu/view/sideMenuView.js (hunk 12 above)
Comment (feruzm) [reply src/components/sideMenu/view/sideMenuView.js]: ok!

File: src/components/sideMenu/view/sideMenuView.js (hunk 9 above)
Comment (feruzm) [reply src/components/sideMenu/view/sideMenuView.js]: ok!

File: src/components/sideMenu/view/sideMenuView.js (hunk 10 above)
Comment (feruzm) [reply src/components/sideMenu/view/sideMenuView.js]: ok!

File: src/screens/application/container/applicationContainer.js (hunk 6 above)
Comment (feruzm) [reply src/screens/application/container/applicationContainer.js]: maybe we should update it on listener when app comes back from background to foreground, do we have any listener like that already? @mistikk

**Comment by feruzm** [comment]
@mistikk fixed all, only 
1. `react native upgrade apply` if you share commands to do it, I could apply changes fully
2. `background to foreground listener` questions left, it makes more sense to add refresh of globalprops in that listener instead of timer listener.
